azurelinuxagent/common/rdma.py

```diff
@@ -199,10 +199,24 @@ def start(self):
     def process(self):
         try:
             RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
-            RDMADeviceHandler.wait_rdma_device(
-                self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
-            RDMADeviceHandler.write_rdma_config_to_device(
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False
+            retcode,out = shellutil.run_get_output("modinfo hv_network_direct")
+            if retcode == 0:
+                version = re.search("version:\s+(\d+)\.(\d+)\.(\d+)\D", out, re.IGNORECASE)
+                if version:
```

Comment:
It's possible version will not show up in older LIS releases, or in a build in kernel. Those are older versions of driver and we simply write to the device to initialize it.